osu.Game.Rulesets.Mania/ManiaInputManager.cs

```diff
@@ -1,21 +1,41 @@
 ﻿// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
 // Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
 
+using System.ComponentModel;
 using osu.Framework.Input.Bindings;
 using osu.Game.Rulesets.UI;
 
 namespace osu.Game.Rulesets.Mania
 {
     public class ManiaInputManager : RulesetInputManager<ManiaAction>
     {
-        public ManiaInputManager(RulesetInfo ruleset)
-            : base(ruleset, 0, SimultaneousBindingMode.Unique)
+        public ManiaInputManager(RulesetInfo ruleset, int variant)
+            : base(ruleset, variant, SimultaneousBindingMode.Unique)
         {
         }
     }
 
     public enum ManiaAction
     {
-        // placeholder
+        [Description("Key 1")]
+        Key1,
+        [Description("Key 2")]
+        Key2,
+        [Description("Key 3")]
+        Key3,
+        [Description("Key 4")]
+        Key4,
+        [Description("Key 5")]
+        Key5,
+        [Description("Key 6")]
+        Key6,
+        [Description("Key 7")]
+        Key7,
+        [Description("Key 8")]
+        Key8,
+        [Description("Key 9")]
+        Key9,
+        [Description("Special")]

```

Comment:
how about making special at the start of the enum? it would not only align the keys to their `int` representation, but also mean that if/when we add more keys they will still be consecutive.

going further, future-proofing for multiple special keys we could start the actual keys at 10 or something, but that's up to you.